### PR TITLE
Improve job keep alive and monitoring logic

### PR DESCRIPTION
**Pull request by tongwu-msft**
## Description
1. Replace GetJobById to GetJobsByIds to reduce DB call
2. Return CancelRequested from KeepHeartbeat to reduce DB call.
3. Only cancel processing jobs for critical error in orchestrator or other processing jobs.

## Related issues
Addresses [issue #].

## Testing
Add& Update unit tests

## FHIR Team Checklist
- [x] **Update the title** of the PR to be succinct and less than 50 characters
- [ ] **Add a milestone** to the PR for the sprint that it is merged (i.e. add S47)
- [x] Tag the PR with the type of update: **Bug**, **Dependencies**, **Enhancement**, or **New-Feature**
- [x] Tag the PR with Azure API for FHIR if this will release to the Azure API for FHIR managed service (CosmosDB or common code related to service)
- [x] Tag the PR with Azure Healthcare APIs if this will release to the Azure Healthcare APIs managed service (Sql server or common code related to service)
- [x] CI is green before merge
- Review [squash-merge requirements](https://github.com/microsoft/fhir-server/blob/master/SquashMergeRequirements.md)

### Semver Change ([docs](https://github.com/microsoft/fhir-server/blob/master/docs/Versioning.md))
Patch|Skip|Feature|Breaking (reason)

### CODE REVIEW

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs [hunk 1]
```diff
@@ -170,6 +170,22 @@ public async Task GivenJobKeepPutHeartbeatWithResult_WhenDequeue_ThenJobWithResu
             Assert.Equal(jobInfo1.Result, jobInfo2.Result);
         }
 
+        [Fact]
+        public async Task GivenRunningJobCanelled_WhenKeepHeartbeat_ThenCancelRequestedShouldBeReturened()
```
Comment (rbans96) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:174]: super nit: GivenRunningJobCan**c**elled_WhenKeepHeartbeat_ThenCancelRequestedShouldBeRetur**n**ed

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs [hunk 2]
```diff
@@ -170,6 +170,22 @@ public async Task GivenJobKeepPutHeartbeatWithResult_WhenDequeue_ThenJobWithResu
...
+        {
+            byte queueType = (byte)TestQueueType.GivenRunningJobCanelled_WhenKeepHeartbeat_ThenCancelRequestedShouldBeReturened;
```
Comment (rbans96) [inline test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:176]: super nit: GivenRunningJobCan**c**elled

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs (hunk 1 above)
Comment (tongwu-msft) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:174]: Thanks updated.

File: test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs (hunk 2 above)
Comment (tongwu-msft) [reply test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/SqlQueueClientTests.cs:176]: Thanks, updated.

File: src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/PutJobHeartbeat.sql [hunk 3]
```diff
@@ -42,6 +45,7 @@ BEGIN TRY
       THROW 50404, 'Job record not found', 1
   END
 
+  Select @CancelRequested
```
[the remark this answers is not in the document]
Comment (tongwu-msft) [reply src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/PutJobHeartbeat.sql:48]: Thanks, updated.

**Review by LTA-Thinking — APPROVED** [review]

**Review by SergeyGaluzo — APPROVED** [review]
:shipit:

**Pull request merged**